Description: Use only stuff included in linux-kbuild
 This patch removes all calls of files that are only included
 in linux-headers during package build time. Since there is
 no single linux-headers package that can be used as
 Build-Depends, the unnecessary calls of files from these 
 packages are removed.
Author: Jan Luca Naumann <j.naumann@fu-berlin.de>
Forwarded: not-needed
Last-Update: 2016-08-31
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Makefile
+++ b/Makefile
@@ -10,6 +10,8 @@
 $(warning could not find kernel config file. internal auto-config may fail)
 endif
 
+DISABLEHEADERDEPS ?= false
+
 CONFIG_AUFS_FS = m
 AUFS_DEF_CONFIG = -DCONFIG_AUFS_FS_MODULE -UCONFIG_AUFS
 include config.mk
@@ -23,7 +25,9 @@
 all: aufs.ko usr/include/linux/aufs_type.h
 
 clean:
+ifneq ($(DISABLEHEADERDEPS),true)
 	${MAKE} ${MakeMod} $@
+endif
 	find . -type f -name '*~' | xargs -r ${RM}
 	${RM} -r aufs.ko usr
 
@@ -45,6 +49,5 @@
 usr/include/linux/aufs_type.h:
 	${MAKE} -rR -C ${KDIR} \
 		-f scripts/Makefile.headersinst \
-		-f Makefile \
 		obj=${d}/include/uapi/linux dst=${d}/usr/include/linux
 	test -s $@
--- a/config.mk
+++ b/config.mk
@@ -49,6 +49,8 @@
 
 ########################################
 
+ifneq ($(DISABLEHEADERDEPS),true)
+
 ifdef CONFIG_AUFS_SBILIST
 ifndef CONFIG_AUFS_MAGIC_SYSRQ
 ifndef CONFIG_PROC_FS
@@ -128,3 +130,5 @@
 $(warning AUFS_BR_FUSE is enabled but FUSE_FS)
 endif
 endif
+
+endif # DISABLEHEADERDEPS
